samza-core/src/main/java/org/apache/samza/execution/JobNode.java

```diff
@@ -130,8 +131,20 @@ public JobConfig generateConfig(String executionPlanJson) {
     Map<String, String> configs = new HashMap<>();
     configs.put(JobConfig.JOB_NAME(), jobName);
 
-    List<String> inputs = inEdges.stream().map(edge -> edge.getFormattedSystemStream()).collect(Collectors.toList());
+    final List<String> inputs = new ArrayList<>();
+    final List<String> broadcasts = new ArrayList<>();
+    for (StreamEdge inEdge : inEdges) {
+      String formattedSystemStream = inEdge.getFormattedSystemStream();
```

Comment:
A couple of reasons why it's done this way:
- broadcast is a valid attribute of stream, so it should be in stream spec. For example, if you think in the future we want to have a input stream as a broadcast stream, we should have the streamspec.broadcast=true. That way planner will put this as a broadcast stream. This does not relate to the broadcast operator.

- broadcast operator provides a way to create an intermediate stream which will be broadcasted to all downstream task operators. Please note this is also an intermediate stream since it's managed by Samza and all the watermark/eos messages will be populated in-band (already tested in beam job and they work fine). This also means we have two kinds of intermediate streams, one created from partitionBy, the other from broadcast.